src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +325,113 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private readonly List<WeakReference<Stats>?>? _allStats;
+        private readonly Stats? _accumulatedStats;
+        private ThreadLocal<Stats>? _stats;
+
+        private (long, long) Sum()
+        {
+            long hits = 0, misses = 0;
+            lock (_allStats!)
+            {
+                if (IntPtr.Size == 4)
```

Comment:
```suggestion
        hits += _accumulatedStats!.Hits;
        misses += _accumulatedStats!.Misses;
```
The local variables are never accessed from concurrent threads and the _accumulatedStats fields are only accessed under the _allStats lock so no need for interlocked here.